File: packages/cli/src/__tests__/fixtures/sample_module.py
"""Sample Python module for entry point testing."""


def process_order(order_id: str) -> dict:
    """Process an order."""
    return {"id": order_id, "status": "processed"}


class OrderService:
    """Service for managing orders."""

    def __init__(self):
        self.orders = []

    def create(self, data):
        self.orders.append(data)


async def async_handler(request):
    """An async request handler."""
    return {"ok": True}

File: packages/cli/src/__tests__/fixtures/sample_module.ts
/**
 * Sample TypeScript module for entry point testing.
 */

export function processOrder(orderId: string): Record<string, string> {
  return { id: orderId, status: 'processed' }
}

export const ORDER_TIMEOUT = 30_000

export class OrderService {
  private orders: string[] = []

  create(data: string): void {
    this.orders.push(data)
  }
}

export async function internalHelper(): Promise<void> {
  // helper for tree-sitter testing
}
